api/database/migrations/2023_09_20_211150_combine_pool_skills.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Query\Expression;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        // create the combined table
+        Schema::create('pool_skill', function (Blueprint $table) {
+            $table->uuid('id')->primary('id')->default(new Expression('gen_random_uuid()'));
+            $table->uuid('skill_id');
+            $table->foreign('skill_id')->references('id')->on('skills');
+            $table->uuid('pool_id');
+            $table->foreign('pool_id')->references('id')->on('pools');
+            $table->unique(['skill_id', 'pool_id']);
+            $table->timestamps();
+            $table->string('type');
+        });
+
+        // fill the new table with pool-skill rows using the prior two tables
+        $essentialPoolSkills = DB::table('pools_essential_skills')->get();
+        foreach ($essentialPoolSkills as $instance) {
+            DB::table('pool_skill')->insert([
+                'skill_id' => $instance->skill_id,
+                'pool_id' => $instance->pool_id,
+                'type' => 'ESSENTIAL',
+            ]);
+        }
+        $nonessentialPoolSkills = DB::table('pools_nonessential_skills')->get();
+        // must not have duplication, pool-skill combo as essential as well as nonessential
+        $filteredNonessential = $nonessentialPoolSkills->reject(function ($value) use ($essentialPoolSkills) {
+            foreach ($essentialPoolSkills as $instance) {
+                if (
+                    $value->skill_id === $instance->skill_id &&
+                    $value->pool_id === $instance->pool_id
+                ) {
+                    return true;
+                }
+            }
+        });
+        foreach ($filteredNonessential as $instance) {
+            DB::table('pool_skill')->insert([
+                'skill_id' => $instance->skill_id,
+                'pool_id' => $instance->pool_id,
+                'type' => 'NONESSENTIAL',
+            ]);
```

Comment:
It would be much faster to insert everything in a single DB operation. Use the loop to assemble an array of `pool_skill` rows, then you only need a single `insert` call. https://laravel.com/docs/10.x/queries#insert-statements

Reply:
This goes for the essential skills too, of course

Reply:
Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7953/commits/a54bbf43eff4167110264b0ff9a3cc8d13b4a006
Running migration forward and back even without a huge amount of rows and there is still a noticeable speed change